src/Analysis/Ast/Impl/Extensions/AstExtensions.cs

```diff
@@ -60,9 +61,28 @@ public static Expression FindExpression(this PythonAst ast, SourceLocation locat
             return ast.CommentLocations[match].Column < location.Column;
         }
 
+        internal static bool HasComment(this PythonAst ast, PythonModule module, string comment, int lineNum) {
```

Comment:
This does not seem right. If you are passing the AST, then it should be using to get locations and not `module.GetLine` since that appears to be using AST attached to the module which is most probably stale.

Reply:
I went ahead and passed the AST to module.GetLine()